Base/QTGUI/qSlicerSaveDataDialog.cxx

```diff
@@ -709,6 +721,105 @@ ctkDirectoryButton* qSlicerSaveDataDialogPrivate::createFileDirectoryWidget(cons
   return directoryButton;
 }
 
+//-----------------------------------------------------------------------------
+void qSlicerSaveDataDialogPrivate
+::makeUserMessages(vtkMRMLStorableNode *node, QWidget *userMessagesWidget)
+{
+  QHBoxLayout * const userMessagesLayout = userMessagesWidget ?
+    dynamic_cast<QHBoxLayout *>(userMessagesWidget->layout()) : nullptr;
+
+  if (!userMessagesLayout)
+    {
+    return;
+    }
+
+  static const QPixmap okayIcon = qApp->style()->standardPixmap(QStyle::SP_DialogApplyButton);
+  static const QPixmap warningIcon = qApp->style()->standardPixmap(QStyle::SP_MessageBoxWarning).scaledToHeight(18);
+  static const QPixmap errorIcon = qApp->style()->standardPixmap(QStyle::SP_MessageBoxCritical).scaledToHeight(18);
+  static const QPixmap questionIcon = qApp->style()->standardPixmap(QStyle::SP_MessageBoxQuestion).scaledToHeight(18);
+
+  const vtkMRMLStorageNode * const snode = node ? node->GetStorageNode() : nullptr;
+
+  // Clear out any existing items in the userMessagesLayout
+  QLayoutItem *layoutItem = nullptr;
+  while ((layoutItem = userMessagesLayout->takeAt(0)) != nullptr)
+    {
+    delete layoutItem;
```

Comment:
Deleting items is tricky and risky. It is better to create them once and just show/hide as appropriate.
For example, if you simply delete `layoutItem` here then its parent may want to delete it again when the save dialog is destroyed (potentially causing crash during application exit).
A simpler solution could be to just show a single icon (maybe on a toolbutton) and just show/hide this button and change the icon as appropriate. Shown only one icon - if there are warning(s) and error(s) then just show a single error icon. If there are only warning(s) then show warning icon.

Reply:
The `takeAt(0)` command is successively removing the currently first widget from the parent.  "[The following code fragment shows a safe way to remove all items from a layout.](https://doc.qt.io/qt-5/qlayout.html#takeAt)"  So, it should all be good.

This late in the game, I'd rather not switch the functionality to showing a single icon.